processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndexCursorFactory.java

```diff
@@ -77,7 +78,20 @@ public IncrementalIndexCursorFactory(IncrementalIndex index)
   @Override
   public CursorHolder makeCursorHolder(CursorBuildSpec spec)
   {
-    return new IncrementalIndexCursorHolder(index, spec);
+    final IncrementalIndexRowSelector rowSelector;
+    final CursorBuildSpec buildSpec;
+    final String timeColumnName;
+    final IncrementalIndex.ProjectionRowSelector projection = index.getProjection(spec);
+    if (projection == null) {
+      rowSelector = index;
+      buildSpec = spec;
+      timeColumnName = ColumnHolder.TIME_COLUMN_NAME;
+    } else {
+      rowSelector = projection.getRowSelector();
+      buildSpec = projection.getRewrittenBuildSpec();
+      timeColumnName = projection.getTimeColumnName();
+    }
+    return new IncrementalIndexCursorHolder(rowSelector, buildSpec, timeColumnName, projection != null);
```

Comment:
In practice I don't think `PreAggregatedIncrementalIndexCursorHolder` makes much difference unless it also makes its own version of `IncrementalIndexCursor` and/or `IncrementalIndexColumnSelectorFactory` which are the real things that get used by the engines. However, implementing these things doesn't really seem necessary, since its all basically just selecting stuff from `FactsHolder` + aggs, which is why I just kept it in the same holder for now, because i think it would basically be just overriding this method